src/components/LocaleSwitcher/LocaleSwitcher.tsx

```diff
@@ -0,0 +1,23 @@
+import { useLocale, useTranslations } from 'next-intl';
+
+import { routing } from '../../i18n/routing';
+import LocaleSwitcherSelect from '../LocaleSwitcherSelect/LocaleSwitcherSelect';
+
+export default function LocaleSwitcher() {
+  const t = useTranslations('LocaleSwitcher');
+  const locale = useLocale();
+
+  return (
+    <LocaleSwitcherSelect defaultValue={locale} label={t('label')}>
+      {routing.locales.map((cur) => (
+        <option
+          key={cur}
+          value={cur}
+          className="bg-background-dark text-foreground-dark dark:bg-background dark:text-foreground"
+        >
+          {t(cur)}
+        </option>
+      ))}
```

Comment:
```suggestion
    <LocaleSwitcherSelect defaultValue={locale} label={t('label')}>
      {routing.locales.map((locale) => (
        <option
          key={locale}
          value={locale}
          className="bg-background-dark text-foreground-dark dark:bg-background dark:text-foreground"
        >
          {t(locale)}
        </option>
      ))}
```

The variable name `cur` is a bit short and isn't immediately clear what it represents. A more descriptive name like `locale` would improve readability.